### PR TITLE
Implements LogarithmicUnits such as dB, dBm, octave or neper

### LINKED ISSUE #71: log and exponential unit conversions
In optics, the power of light is often measured in dBm. The definition of dBm is:

dBm = 10 \* log(mW)

It means that 0 dBm is 1 mW, 10 dBm is 10 mW and -10 dBm is 0.1 mW.

The opposite conversion is:

mW = 10**(dBm / 10)

### PR DESCRIPTION
**Pull request by 5igno**
- [X] Closes #71
- [X] Executed ``black -t py36 . && isort -rc . && flake8`` with no errors
- [x] The change is fully covered by automated unit tests
- [ ] Documented in docs/ as appropriate
- [X] Added an entry to the CHANGES file

### CODE REVIEW

File: pint/compat.py [hunk 1]
```diff
@@ -11,8 +11,14 @@
 import tokenize
 from decimal import Decimal
 from io import BytesIO
+from math import exp as math_exp
+from math import log as math_log
 from numbers import Number
 
+# Defines Logarithm and Exponential for Logarithmic Converter
+from numpy import exp as np_exp
```
Comment (hgrecco) [inline pint/compat.py]: This is not a good idea, because numpy might not be installed

File: pint/default_en.txt [hunk 2]
```diff
@@ -174,7 +172,7 @@ week = 7 * day
 fortnight = 2 * week
 year = 365.25 * day = a = yr = julian_year
 month = year / 12
-decade = 10 * year
+# decade = 10 * year
```
Comment (hgrecco) [inline pint/default_en.txt]: Add a note here indicating why this is commented. (We will have to make a decision about this and `B`)

File: pint/default_en.txt [hunk 3]
```diff
@@ -467,6 +466,30 @@ buckingham = debye * angstrom
...
+# Logaritmic Units of dimensionless quantity: [ https://en.wikipedia.org/wiki/Level_(logarithmic_quantity) ]
+
+decibell = 1 ; logbase: 10; logfactor: 10 = dB
+# bell = 1 ; logbase: 10; logfactor: = B
+## NOTE: B (Bell) symbol conflicts with byte
+##       Defining dB as deci- Bell causes bad assignment of the unit to ScaleConverter, with poor results
+
+decade = 1 ; logbase: 10; logfactor: 1 
+
+neper = 1 ; logbase: 2.71828182845904523536028747135266249775724709369995; logfactor: 0.5 = Np
```
Comment (hgrecco) [inline pint/default_en.txt]: Do not put here the `2.71...`, use the constant defined in `contants`

Comment (5igno) [reply pint/default_en.txt]: I defined `eulers_number` in `constants` [ [ee0a4cb](https://github.com/hgrecco/pint/pull/1024/commits/ee0a4cb29494d5a4c0453c8785ef992ceeff36f2) ], but parsing constants within the modifier raises an error:
```
AssertionError: 1 != 0 : 1 warnings raised.
E   In line 483, cannot add 'neper = 1 ; logbase: eulers_number; logfactor: 0.5 = Np' Unit definition ('neper') must contain only numbers in modifier, not  eulers_number
```

File: pint/compat.py (hunk 1 above)
Comment (5igno) [reply pint/compat.py]: I agree, it's not great. But I was getting a warning of unused libraries by flake8.
Any idea of how to get around this issue?

File: pint/default_en.txt (hunk 3 above)
Comment (5igno) [reply pint/default_en.txt]: I forgot in the commit the neper's definiton with eulers_number: you can see in Travis that most tests fail :(

File: pint/compat.py (hunk 1 above)
Comment (5igno) [reply pint/compat.py]: OK, I found a way to solve this: you can find it [at this commit](https://github.com/hgrecco/pint/pull/1024/commits/fcab800d3e6267fa6f2c13f3dfa4e0cb192827c4)

**Comment by hgrecco** [comment]
Please remove all new lines added in docstrings or `if`  and try to keep your changes only to the feature you are implementing. This makes reviewing easier. You can stash all cosmetic and typo mistakes so we can discuss them in a later stage.

File: pint/default_en.txt (hunk 3 above)
Comment (5igno) [reply pint/default_en.txt]: If magic numbers in the definition are a problem, how would you suggest to proceed? Should I remove neper's definition from `default-en.txt`?

**Comment by 5igno** [comment]
I will rearrange the changes so that the pull request is easier.
Sorry for the messy pull request, and thank you for your patience.

**Pull request closed without merging**